lib/shopify_api/resources/product.rb

```diff
@@ -29,5 +29,42 @@ def add_to_collection(collection)
     def remove_from_collection(collection)
       collection.remove_product(self)
     end
+
+    def self.search(params)
```

Comment:
This will be kinda slow and I think you'll be limited to a single page of products. The function is also super massive and doesn't feel right.

I think we can perhaps take a better approach to this or simply make the API respond to more useful queries. If we can't take the query approach then let's fix up this function to be a bit easier to read, faster, etc.